### PR TITLE
chore(typescript): Update typescript and fix related build errors to

### PR DESCRIPTION
**Pull request by SimonSchick**
Update to typescript 4.2.1
`@Column` decorator now takes a single argument instead of a vararg to work around a build error
`Model.on<T>` now restricts on `object`
Changed type of `filter` property on `IQueryOptions` in order to fix a build error and ensuring type safety for logical expressions
Adde various generic casts to fix build
Remove LazySubject as it is unused

BREAKING CHANGE: The `@Column` decorator now only takes one argument

Before: `@Decorate({ a: 1 } , DateAccessor)`
After: `@Decorator({ a: 1, ...DateAccessor })`

BREAKING CHANGE: The `LazySubject` class has been removed.

BREAKING CHANGE: `IQueryOptions.filter` has a new type in order to fix type ambigiuty due to index signature problems.

Before:
`{ filter: { someProperty: 1, $or: [{ a: 1 }, { b: 1}] } }`
After:
`{ filter: { $and: [{ someProperty: 1 }, { $or: [{ a: 1 }, { b: 1 }] }] } }`

### CODE REVIEW

**Comment by ProbablePrime** [comment]
`bugfix` isn't a valid prefix: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#type

**Comment by SimonSchick** [comment]
@ProbablePrime is there a bot that can do a validation for this?

**Comment by SimonSchick** [comment]
😱

**Comment by UnwrittenFun** [comment]
there seems to be some neat tools around it
https://github.com/willsoto/validate-commit
https://www.npmjs.com/package/commitizen

**Comment by ProbablePrime** [comment]
There's a tool i saw at some point

**Comment by ProbablePrime** [comment]
I was laying in bed reading reddit and saw something cool here :( but now i cant find it. Google has a ton though.

**Comment by ProbablePrime** [comment]
Found it: https://activeviam.com/blog/setting-up-and-deploying-a-github-app-in-the-azure-cloud/

It stuck out because it uses azure functions(msft flavor aws lambda)

File: src/Types.ts [hunk 1]
```diff
@@ -67,7 +69,7 @@ export interface IQueryOptions extends IExtraOptions {
     /**
      * filter specifies an optional list of filters to apply to the query.
      */
-    filter?: { [attribute: string]: WhereClause };
+    filter?: { [attribute: string]: Comparator } | IConjunction | IDisjunction;
```
Comment (UnwrittenFun) [inline src/Types.ts]: This could be `filter?: WhereClause;`

File: src/adapters/components/QueryBuilder.ts [hunk 2]
```diff
@@ -19,8 +18,12 @@ export interface IAPIQueryParameters {
     noCount?: number;
 }
 
-function isDisjunction(val: WhereClause, key: string): val is IDisjunctionArgument[] {
-    return val instanceof Array && key === '$or';
+function isDisjunction(val: any): val is IDisjunction {
+    return val.$or instanceof Array;
```
Comment (UnwrittenFun) [inline src/adapters/components/QueryBuilder.ts]: Should probably be `Array.isArray(val.$or)`

**Pull request merged**